cpu/sam0_common/periph/uart.c

```diff
@@ -368,7 +368,34 @@ int uart_mode(uart_t uart, uart_data_bits_t data_bits, uart_parity_t parity,
 
     return UART_OK;
 }
-#endif
+#endif /* MODULE_PERIPH_UART_MODECFG */
+
+#ifdef MODULE_PERIPH_UART_RXSTART_IRQ
+void uart_rxstart_irq_configure(uart_t uart, uart_rxstart_cb_t cb, void *arg)
+{
+    /* CTRLB is enable-proteced */
+    dev(uart)->CTRLA.bit.ENABLE = 0;
+
+    /* set start of frame detection enable */
+    dev(uart)->CTRLB.reg |= SERCOM_USART_CTRLB_SFDE;
+
+    uart_ctx[uart].rxs_cb  = cb;
+    uart_ctx[uart].rxs_arg = arg;
+
+    /* enable UART again */
+    dev(uart)->CTRLA.bit.ENABLE = 1;
+}
+
+void uart_rxstart_enable(uart_t uart)
+{
+    dev(uart)->INTENSET.reg = SERCOM_USART_INTENSET_RXS;
```

Comment:
I think, we should clear pending IRQs. If I understand the datasheet correctly, the ISR would fire immediately after enabling the IRQ.

Reply:
Since you are clearing pending IRQs in the first lines of the ISR, this shouldn't be necessary, right?